autoPyTorch/evaluation/tae.py

```diff
@@ -62,34 +91,150 @@ def fit_predict_try_except_decorator(
+
+    _success_in_anything_exc = (is_anything_exception and obj.exit_status == 0)
+    _success_in_to_or_mle = (status in [StatusType.SUCCESS, StatusType.DONOTADVANCE]
+                             and not is_anything_exception)
+
+    if _success_in_anything_exc or _success_in_to_or_mle:
+        cost = result
+    if not is_anything_exception or not _success_in_anything_exc:
+        additional_run_info.update(
+            subprocess_stdout=obj.stdout,
+            subprocess_stderr=obj.stderr,
+            info=info_msg)
+    if is_anything_exception and not _success_in_anything_exc:
+        status = StatusType.CRASHED
+        additional_run_info.update(exit_status=_encode_exit_status(obj.exit_status))
+
+    return cost, status, info, additional_run_info
+
+
+def _process_exceptions(
+    obj: PynisherFunctionWrapperType,
+    queue: Queue,
+    budget: float,
+    worst_possible_result: float
+) -> ProcessedResultsType:
+    if obj.exit_status is TAEAbortException:
+        info, status, cost = None, StatusType.ABORT, worst_possible_result
+        additional_run_info = dict(
+            error='Your configuration of autoPyTorch did not work',
+            exit_status=_encode_exit_status(obj.exit_status),
+            subprocess_stdout=obj.stdout,
+            subprocess_stderr=obj.stderr
+        )
+        return cost, status, info, additional_run_info
+
+    info_for_empty: Dict[str, Any] = {}
+    if obj.exit_status in (pynisher.TimeoutException, pynisher.MemorylimitException):
+        is_timeout = obj.exit_status is pynisher.TimeoutException
+        status = StatusType.TIMEOUT if is_timeout else StatusType.MEMOUT
+        is_anything_exception = False
+        info_msg = f'Run stopped because of {"timeout" if is_timeout else "memout"}.'
+        info_for_empty = {'error': 'Timeout' if is_timeout else 'Memout'}
+    else:
+        status, is_anything_exception = StatusType.CRASHED, True
+        info_msg = 'Run treated as crashed because the pynisher exit ' \
+                   f'status {str(obj.exit_status)} is unknown.'
+        info_for_empty = dict(
+            error='Result queue is empty',
+            exit_status=_encode_exit_status(obj.exit_status),
+            subprocess_stdout=obj.stdout,
+            subprocess_stderr=obj.stderr,
+            exitcode=obj.exitcode
+        )
+
+    cost, status, info, additional_run_info = _exception_handling(
+        obj=obj, queue=queue, is_anything_exception=is_anything_exception,
+        info_msg=info_msg, info_for_empty=info_for_empty,
+        status=status, worst_possible_result=worst_possible_result
+    )
+
+    if budget == 0 and status == StatusType.DONOTADVANCE:
+        status = StatusType.SUCCESS
+
+    if not isinstance(additional_run_info, dict):
+        additional_run_info = {'message': additional_run_info}
+
+    return cost, status, info, additional_run_info
+
+
+class TargetAlgorithmQuery(AbstractTAFunc):
```

Comment:
I think `ExecuteTAFuncWithQueue` was more appropriate as a name. Also, it keeps `AutoPyTorch` in line with `SMAC` and `auto-sklearn`.  Let's keep it that way.

Reply:
Since this is not urgent, I will just reply to this message and another for now.

Although I agree that `TargetAlgorithmQuery` is not compatible with `auto-sklearn`, this name is still compatible with `SMAC` and the name itself is more appropriate because of the following problems in the previous name.

The problems of `ExecuteTAFuncWithQueue` are that:
1. the name starts from a verb, which is confusing for a class name except the class is callable,
2. the name itself is not correct because we **execute TA func** when we **query** it from **Queue**, but we never execute TA func with Queue,
3. instances of this class are stored in Queue as a query, so it will not be executed when it is instantiated

It is impossible to merge `BaseTask` with the one in ASK, so I did not feel the compatibility with ASK in `TAE` is necessary, but do you think we need it?